### PR TITLE
Internationalization of dnf-plugin-spacewalk

### PR DESCRIPTION
**Pull request by hustodemon**
## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: i18n

- [x] **DONE**

## Test coverage
- No tests: i18n

- [x] **DONE**

## Links

https://github.com/SUSE/spacewalk/issues/12328

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by hustodemon** [comment]
@mcalmer IIUC, it's enough to include the `pot` file. The `po` for each supported language will be brough by weblate. Is that correct?

**Comment by mcalmer** [comment]
> @mcalmer IIUC, it's enough to include the `pot` file. The `po` for each supported language will be brough by weblate. Is that correct?

yes

**Comment by mcalmer** [comment]
But you should "install" all translations. So check the spec file if it already collect language files.

**Comment by hustodemon** [comment]
Thanks @mcalmer , will do.

File: client/rhel/dnf-plugin-spacewalk/po/dnf-plugin-spacewalk.pot [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2020-11-05 13:12+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=CHARSET\n"
```
Comment (parlt91) [inline client/rhel/dnf-plugin-spacewalk/po/dnf-plugin-spacewalk.pot:17]: I am not sure if weblate generates the charset for the `po` files from the `pot` file. But just to be sure could you `charset=UTF-8`?

**Review by parlt91 — APPROVED** [review]
Thanks. LGTM
Just one comment below

**Comment by mbologna** [comment]
I am a bit confused: for other PRs we also have a Makevars file to declare the ugettext function:
https://github.com/uyuni-project/uyuni/pull/2650/files#diff-aa16b3e3d95f1d23b6c0fab8ab77a6cfb91a9b22901273ccc0b5d7f4cf79cf82
but in dnf-plugin-spacewalk I cannot find any gettext hook imported.

**Comment by hustodemon** [comment]
@mbologna True, but where is this file used? Weblate? I couldn't find any usage in uyuni.

**Comment by mbologna** [comment]
IIRC the Makevars instructs gettext about the method name (`_`) that is used in gettext calls

**Comment by hustodemon** [comment]
Thanks @mbologna it's [true indeed](https://www.gnu.org/software/gettext/manual/html_node/po_002fMakevars.html). Although, according to gettext manual, the `_` seems to be explicitly used for python, it doesn't hurt to explicitly add it. We also do that for some other python projects in Uyuni (`spacecmd`).

**Comment by mcalmer** [comment]
If `Makevars` is needed or not also depends on how the Makefile was written. Not sure where this was copied from.
We have also examples in this git where no Makevars file is needed.

File: client/rhel/dnf-plugin-spacewalk/po/dnf-plugin-spacewalk.pot (hunk 1 above)
Comment (hustodemon) [reply client/rhel/dnf-plugin-spacewalk/po/dnf-plugin-spacewalk.pot:17]: It seems `charset=CHARSET` is common for our `pot` files. I'd stick to that.

**Comment by hustodemon** [comment]
@mcalmer The makefile was copied from other project from `client/` dir (I think it was `yum-rhn-plugin`). But the `Makefile` doesn't need to specify `Makevars` file - the file is taken into account even without it (I've tried running running `make update-po` while changing the `Makevars` content and the results differed).

**Pull request merged**